app/src/test/kotlin/xyz/block/augurref/api/FeeEstimateEndpointTest.kt

```diff
@@ -0,0 +1,250 @@
+    every { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(3.0) } returns mockFeeEstimate
+
+    configureTestApplication()
+
+    client.get("/fees/target/3").apply {
+      assertEquals(HttpStatusCode.OK, status)
+      val responseBody = bodyAsText()
+      val response: FeeEstimateResponse = objectMapper.readValue(responseBody)
+
+      assertEquals(fixedInstant, response.mempoolUpdateTime)
+      assertNotNull(response.estimates)
+    }
+
+    verify { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(3.0) }
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = ["abc"])
+  fun `should return 400 for invalid num_blocks parameter`(invalidValue: String) = testApplication {
+    configureTestApplication()
+
+    client.get("/fees/target/$invalidValue").apply {
+      assertEquals(HttpStatusCode.BadRequest, status)
+      assertEquals("Invalid or missing number of blocks", bodyAsText())
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = ["-1", "0"])
+  fun `should handle edge case num_blocks parameters`(edgeValue: String) = testApplication {
+    val numBlocks = edgeValue.toDouble()
+    // Mock return null for edge cases (could be valid behavior)
+    every { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(numBlocks) } returns null
+
+    configureTestApplication()
+
+    client.get("/fees/target/$edgeValue").apply {
+      assertEquals(HttpStatusCode.ServiceUnavailable, status)
+      assertEquals("No fee estimates available yet", bodyAsText())
+    }
+
+    verify { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(numBlocks) }
+  }
+
+  @Test
+  fun `should return 404 for empty num_blocks parameter`() = testApplication {
+    configureTestApplication()
+
+    // Empty string in URL path results in /fees/target/ which doesn't match the route
+    client.get("/fees/target/").apply {
+      assertEquals(HttpStatusCode.NotFound, status)
+    }
+  }
+
+  @Test
+  fun `should return 503 when no estimates available for target`() = testApplication {
+    every { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(2.0) } returns null
+
+    configureTestApplication()
+
+    client.get("/fees/target/2").apply {
+      assertEquals(HttpStatusCode.ServiceUnavailable, status)
+      assertEquals("No fee estimates available yet", bodyAsText())
+    }
+
+    verify { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(2.0) }
+  }
+
+  @Test
+  fun `should handle decimal num_blocks parameter`() = testApplication {
+    val fixedInstant = Instant.parse("2025-01-15T08:15:30.999Z")
+    val mockFeeEstimate = createMockFeeEstimate(fixedInstant)
+    every { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(1.5) } returns mockFeeEstimate
+
+    configureTestApplication()
+
+    client.get("/fees/target/1.5").apply {
+      assertEquals(HttpStatusCode.OK, status)
+    }
+
+    verify { mockMempoolCollector.getLatestFeeEstimateForBlockTarget(1.5) }
+  }
```

Comment:
We shouldn't be able to get block estimates for less than 3 btw

We discussed it in this convo
https://github.com/block/bitcoin-augur/pull/2#discussion_r2205833772

ik the test passes, but from `bitcoin-augur` it should throw an error.

Reply:
nice catch! I'm going to remove that test entirely. Adding validations in the endpoint doesn't make a ton of sense as we might change this in the underlying library.